### ISSUE TITLE
Blacklist for armor and tools being requested

### ISSUE THREAD

**Issue by Saereth** [ldtteam/minecolonies-features#693]
### Prerequisites 

- [x ] I am running the latest **alpha** version of MineColonies for my Minecraft version.
- [x ] I checked the MineColonies/Structurize [wiki](https://wiki.minecolonies.ldtteam.com/) and made sure that my feature does not already exist.
- [x ] I made sure that **this feature request is not a duplicate of any existing feature request.**

### Context 

- Minecraft Version: 1.18.2
- MineColonies Version: minecolonies-1.18.2-1.0.872-BETA

## Is your feature request related to a problem?

Guards currently request modded items that are not craftable, or never craftable which is causing confusion for players. This
is especially prevelant in a highly modded pack. 

## Feature description

A blacklist or a system of adding a specific tag to items that would exclude them from being requested by citizens as armor, weapons or tools would be useful. Even somethin as tag:excluded_from_requests would work wonders to clean up the odd requests in a modded pack. Alternative a whitelisting method would work as well. 

### Notes or related things

Just as an aside, The ability to override the enchantment level of the item counting towards its vanilla tiering would be nice as well as many modded enchants are functionally not power boost and yet still increase the tier. 

---
## Viewers

* Add a thumbs-up to the feature request to vote for it. This helps the request become more visible to the team and doesn't clutter the comments.
* Add a comment if you have any insights or background information that isn't already part of the conversation.

**Comment by Moobien** [comment]
They actually ask for ANY tool/weapon/armor that matches the requested level. They do not and have never asked for specific things.

**Comment by Saereth** [comment]
They cycle through specific things in the fulfillment window and if a player just clicks on them, they initially appear to ask for specific things. For example, the t1 archers will request an enchanted bow from ars. The player nor the mine colonists can craft this item, yet it is what they are asking for by default in this pack. With no way to blacklist that item, it's been very confusing for the players. I've set up a quest explaining they can take other item types, but there is still a lot of confusion regarding it and it's something an exclude tag would solve.

**Comment by Moobien** [comment]
Too many items to exclude. Devs would have to tag every weapon from every mod. It is not realistic to do. It simply cycles through all possible items that can be used to fulfil the request. If people get confused it isnt hard to jump onto the wiki and learn more.

**Comment by MotionlessTrain** [comment]
I think a blacklist tag would be easier than a whitelist tag. That means the team doesn't need to tag everything (as everything is included by default, like it is now), but mod developers or modpack developers can blacklist their items if they are not meant for minecolonies' citizens

**Comment by Saereth** [comment]
yes exactly, and it isnt that much work to tag a bunch of items for pack devs. If I need to blacklist a a dozen and a half weapons I put them all in inventory, dump my inventory with toolkit, paste that into a kubejs array and then iterate over the array to add the tags, all in all it would take like 5mins or so to blacklist every item in the pack that needs it if the tag were simply implemented on minecolony's side.

**Comment by Moobien** [comment]
Would end up bein up to pack makers then.

**Comment by Saereth** [comment]
Why would this be better as an addon? There is no APi support for this that I see, so I'd have to mixin your class just to add a simple blacklisting tag. It's like 4 lines of code to do internally for much-needed optional functionality addition when using the mod in a heavily modded pack.
